misk/src/main/kotlin/misk/client/HttpClientsConfig.kt

```
@@ -1,115 +1,75 @@
-  override fun getInputType(typeFactory: TypeFactory) =
-      typeFactory.constructType(BackwardsCompatibleEndpointConfig::class.java)
-
-  override fun getOutputType(typeFactory: TypeFactory) =
-      typeFactory.constructType(HttpClientEndpointConfig::class.java)
-
-  override fun convert(value: BackwardsCompatibleEndpointConfig) =
-      //http://%s.%s.gns.square
-      if (value.clientConfig == null)
-        HttpClientEndpointConfig(
-            url = value.url,
-            envoy = value.envoy,
-            clientConfig = HttpClientConfig(
-                value.connectTimeout,
-                value.writeTimeout,
-                value.readTimeout,
-                value.pingInterval,
-                value.callTimeout,
-                value.maxRequests,
-                value.maxRequestsPerHost,
-                value.maxIdleConnections,
-                value.keepAliveDuration,
-                value.ssl
-            )
-        )
-      else
-        HttpClientEndpointConfig(
-            url = value.url,
-            envoy = value.envoy,
-            clientConfig = value.clientConfig
-        )
-}
-
+@JsonDeserialize(converter = BackwardsCompatibleClientsConfigConverter::class)
 data class HttpClientsConfig(
-  val defaultConnectTimeout: Duration? = null,
-  val defaultWriteTimeout: Duration? = null,
-  val defaultReadTimeout: Duration? = null,
-  val ssl: HttpClientSSLConfig? = null,
-  val defaultPingInterval: Duration? = null,
-  val defaultCallTimeout: Duration? = null,
-  @JsonDeserialize(contentConverter = BackwardsCompatibleEndpointConfigConverter::class)
+  @JsonAlias("hosts")
+  //Need to retain ordering, hence LinkedHashMap
+  val hostConfigs: LinkedHashMap<String, HttpClientConfig> = linkedMapOf(),
   val endpoints: Map<String, HttpClientEndpointConfig> = mapOf()
 ) : Config {
+  init {
+    validatePatterns()
```

This is to fail fast if configuration has an incorrect regex specified.